### DISCUSSION TITLE
Native async json-rpc

### DISCUSSION THREAD

**Discussion by yyoncho** [emacs-lsp/lsp-mode#3770] [Ideas]
For better visibility of native async jsonrpc implementation in core.

Repo: https://github.com/emacs-lsp/emacs

**Comment by zbelial** [comment]
Hi, have you considered implementing json-rpc as a module? IMO people can benefit from it more soon than the native async json-rpc. 

FYI, I've implemented a LSP client for Emacs using rust, you can find it [here](https://github.com/zbelial/lspce).  From my experience, implementing json-rpc as a module is valuable.
